flink-runtime/src/main/java/org/apache/flink/runtime/security/token/hadoop/HadoopFSDelegationTokenProvider.java

```diff
@@ -110,7 +110,10 @@ public ObtainedDelegationTokens obtainDelegationTokens() throws Exception {
         return freshUGI.doAs(
                 (PrivilegedExceptionAction<ObtainedDelegationTokens>)
                         () -> {
-                            Credentials credentials = new Credentials();
+                            LOG.debug("Loading delegation tokens available to UGI current user");
+                            Credentials credentials =
+                                    new Credentials(
+                                            UserGroupInformation.getCurrentUser().getCredentials());
```

Comment:
Adding current user here would change the behaviour so that Hadoop FS provider would send out Kudu or any other tokens to providers which Hadoop supports. That's not something what I'm comfortable with based on my current knowledge.

Reply:
Understand. How about define a new CurrentUserDelegationTokenProvider mainly for refresh the tokens added into container launch context? They are useful as these tokens are returned for setup execution context for containers. When a TM failure or JM recover from HA mode, these tokens need to be still valid.